src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -5598,22 +5762,37 @@ subroutine atm_compute_solve_diagnostics_work(nCells, nEdges, nVertices, &
       !
       ! Replace 2.0 with 2 in exponentiation to avoid outside chance that
       ! compiler will actually allow "float raised to float" operation
+      MPAS_ACC_TIMER_START('atm_compute_solve_diagnostics [ACC_data_xfer]')
+      !$acc enter data create(ke)
+      MPAS_ACC_TIMER_STOP('atm_compute_solve_diagnostics [ACC_data_xfer]')
+      !$acc parallel default(present)
+      !$acc loop gang
       do iCell=cellStart,cellEnd
-         ke(1:nVertLevels,iCell) = 0.0
+         !$acc loop vector
+         do k=1,nVertLevels
+            ke(k,iCell) = 0.0_RKIND
+         end do
+         !$acc loop seq
          do i=1,nEdgesOnCell(iCell)
             iEdge = edgesOnCell(i,iCell)
+            !$acc loop vector
             do k=1,nVertLevels
 !               ke(k,iCell) = ke(k,iCell) + 0.25 * dcEdge(iEdge) * dvEdge(iEdge) * u(k,iEdge)**2
                ke(k,iCell) = ke(k,iCell) + 0.25 * ke_edge(k,iEdge)
             end do
          end do
 !DIR$ IVDEP
+         !$acc loop vector
          do k=1,nVertLevels
             ke(k,iCell) = ke(k,iCell) * invAreaCell(iCell)
          end do
       end do
+      !$acc end parallel
 
 
+      MPAS_ACC_TIMER_START('atm_compute_solve_diagnostics [ACC_data_xfer]')
+      !$acc enter data create(ke_vertex)
```

Comment:
Since `ke_vertex` is only used inside the `hollingsworth` conditional block, we could move the creation of this variable inside the block. We can also just delete `ke_vertex` at the end of the block, since it isn't used elsewhere in the solver.

Reply:
Addressed by [85d339a](https://github.com/MPAS-Dev/MPAS-Model/pull/1192/commits/85d339a91d026dc67ae7ae917f5e630ccc6634c7)